ui/StatusQ/include/StatusQ/SingleModelItemData/singleModelItemData.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "singleItemFilter.h"
+
+#include <QAbstractItemModel>
+#include <QQmlPropertyMap>
+#include <QPointer>
+#include <QQmlEngine>
+
+
+class SingleModelItemData: public QObject
+{
+    Q_OBJECT
+    // input
+    Q_PROPERTY(QAbstractItemModel* sourceModel READ sourceModel WRITE setSourceModel NOTIFY sourceModelChanged REQUIRED)
+    Q_PROPERTY(SingleItemFilter* filter READ filter WRITE setFilter NOTIFY filterChanged REQUIRED)
+
+    // output
+    Q_PROPERTY(QQmlPropertyMap* modelItem READ item NOTIFY itemChanged)
+    Q_PROPERTY(bool available READ available NOTIFY availableChanged)
+    Q_PROPERTY(QStringList roles READ roles NOTIFY rolesChanged)
+    Q_PROPERTY(int row READ row NOTIFY rowChanged)
+    //Q_PROPERTY(bool cacheOnRemoval READ cacheOnRemoval WRITE setCacheOnRemoval NOTIFY cacheOnRemovalChanged)
+
+public:
+    explicit SingleModelItemData(QObject* parent = nullptr);
+
+    QQmlPropertyMap* item() const;
+    bool available() const;
+    QAbstractItemModel* sourceModel() const;
+    SingleItemFilter* filter() const;
+    const QStringList& roles() const;
+    int row() const;
+
+protected:
+    void setSourceModel(QAbstractItemModel* sourceModel);
+    void setFilter(SingleItemFilter* filter);
+    void setIndex(const QModelIndex& index);
+    void setAvailable(bool available);
+    void setRoles(const QStringList& roles);
+
+    void resetIndex();
+    void tryItemResetOrUpdate();
+    void resetItem();
+    void updateItem(QList<int> roles = {});
```

Comment:
```suggestion
    void updateItem(const QList<int>& roles = {});
```

Reply:
I need a copy in the cpp to handle the empty roles case. I'm replacing empty with all roles